runtime/autoload/provider.vim

```diff
@@ -0,0 +1,18 @@
+" Common functionality for providers
+
+let s:stderr = {}
+
+function! provider#stderr_collector(chan_id, data, event) dict
```

Comment:
Yes, it should be.  The lack of the `dict` attribute on callback functions was one of the problems people had to fix when we [merged partials support](https://github.com/neovim/neovim/wiki/Following-HEAD#20161212).  The issue that needs to be figured out is why this _isn't_ being called with as a dict-function, since it should always be called as a job callback.